### PR TITLE
Added an API to allow oversize protos when using C++ extension in Python

### PR DESCRIPTION
**Pull request by keveman**

### CODE REVIEW

File: python/google/protobuf/internal/oversize_protos_test.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  import unittest
+from google.protobuf import descriptor_pb2
+from google.protobuf import proto_builder
+from google.protobuf.internal import api_implementation
+from google.protobuf.message import DecodeError
+
+
+@unittest.skipIf(api_implementation.Type() != 'cpp',
+                 'explicit tests of the C++ implementation')
+class OversizeProtosTest(unittest.TestCase):
```
Comment (haberman) [inline python/google/protobuf/internal/oversize_protos_test.py]: Please add this class in an existing file, like message_test.py. Our open-sourcing process makes adding new files a big pain, and there isn't much code here.

**Comment by xfxyjwf** [comment]
@haberman @anandolee

File: python/google/protobuf/internal/oversize_protos_test.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          type: TYPE_MESSAGE
+          type_name: "msg1"
+        }
+      }
+    """
+
+  def testAssertOversizeProto(self):
+    from google.protobuf.pyext._message import SetAllowOversizeProtos
+    SetAllowOversizeProtos(False)
+    proto_cls = proto_builder.MakeProtoClassFromFileDescriptor(
```
Comment (haberman) [inline python/google/protobuf/internal/oversize_protos_test.py]: Instead please use an interface like DescriptorPool or Descriptor. The proto_builder interface is kind of a random interface with few users that I'd like to deprecate if possible.

**Comment by haberman** [comment]
Change looks great in general! I just had a few comments.

**Comment by haberman** [comment]
ok to test

**Comment by haberman** [comment]
Whoops, never mind that Jenkins failure, looks like we still have some bugs to iron out in that.

**Comment by haberman** [comment]
retest this please

**Comment by keveman** [comment]
Whoops, looks like we lost some of your comments because I squashed my commits into one. But anyways, I addressed both of your comments : 

> Instead please use an interface like DescriptorPool or Descriptor. The proto_builder interface is kind of a random interface with few users that I'd like to deprecate if possible.

> Please add this class in an existing file, like message_test.py. Our open-sourcing process makes adding new files a big pain, and there isn't much code here.

PTAL.

**Comment by haberman** [comment]
LGTM, thanks. Will just wait for the tests to complete before merging.

**Comment by haberman** [comment]
retest this please

**Comment by haberman** [comment]
(that comment was for Jenkins btw. :)

**Comment by haberman** [comment]
Congratulations, you are the first beneficiary of a sub-10 minute Jenkins build!

I'll merge this now, since I don't foresee any platform-specific breakage on this.

**Pull request merged**